bip-0360.mediawiki

```diff
@@ -1,705 +1,401 @@
 
-We will first look at our approach to the problem of PQ signatures and then give our solution for public keys larger than 520 bytes.
+The control block is a 1 + 32*d byte array, where the first byte is the control byte and the next 32*d bytes are the Merkle path to the leaf script, where d is the depth of the leaf script in the Merkle tree. The control byte is the same as the control byte in a P2TR control block, including the 7 bits which are used to specify the tapleaf version. The parity bit of the control byte is always 1, since P2TSH does not have a key-path spend. Unlike P2TR, we omit the public key from the control block as it is not needed in P2TSH. We maintain support for the optional annex in the witness (see Specification section below for more details).
 
-To keep P2QRH small and simple, we have opted not to raise the stack element size limit as part of P2QRH, but instead make this change when 
-adding PQ signatures. That said, we are not strongly opposed to putting this increase in P2QRH.
-
-We propose a stack element size limit of 8,000 bytes. We arrive at 8,000 by rounding up from the needed 7,856 bytes. 
+==Rationale==
 
-OP_DUP will duplicate any stack element. Thus, if we allowed OP_DUP to duplicate stack elements of size 8,000 bytes, it would be possible 
-to write a tapscript which will duplicate stack elements until it reaches the maximum number of elements on stack, i.e. 1000 elements. 
-An increase from 520 bytes to 8,000 bytes would increase the memory footprint from 520 KB to 8 MB.
+In this proposal, we adopt a “prepared not scared” approach to the possible advancement of quantum computing – and offer Bitcoin users an option for increased protection if they so choose.
 
-To prevent OP_DUP from creating an 8 MB stack by duplicating stack elements larger than 520 bytes we define OP_DUP to fail on stack 
-elements larger than 520 bytes. Note this change to OP_DUP is not consensus critical and does not require any sort of fork. This is
-because currently there is no way to get a stack element larger than 520 bytes onto the stack so triggering this rule is currently
-impossible and would only matter if the stack element size limit was raised.
+We understand that this subject is controversial and veiled in uncertainty; and hope to offer users flexible options - tailored to their level of concern - that are relatively unobtrusive to existing network architecture.
 
-==== Public keys larger than 520 bytes ====
+Design of the P2TSH output type is guided by the following intentions:
 
-Turning our attention to public keys larger than 520 bytes. This is not needed for SLH-DSA as its public key is only 32 bytes. 
-This is a different problem than signatures as public keys are typically pushed onto
-the stack by the script (redeem script) to commit to public keys in the output. The OP_PUSHDATA opcode in tapscript fails if asked to push
-more than 520 bytes onto the stack. 
+<ol>
```

Comment:
Is using HTML in markdown BIPs like this OK? Shouldn't we using markdown for this?